isis/src/control/objs/BundleUtilities/BundleObservation.cpp

```diff
@@ -783,6 +784,785 @@ namespace Isis {
   }
 
 
+
+  /**
+   * @brief Takes in an open std::ofstream and writes out information which goes into the
+   * bundleout.txt file.
+   * @param fpOut The open std::ofstream object which is passed in from
+   * BundleSolutionInfo::outputText()
+   * @param errorPropagation Boolean indicating whether or not to attach more information
+   *     (corrections, sigmas, adjusted sigmas...) to the output.
+   */
+
+  void BundleObservation::bundleOutput(std::ofstream &fpOut, bool errorPropagation) {
+
+      char buf[4096];
+
+      std::vector<double> coefX;
+      std::vector<double> coefY;
+      std::vector<double> coefZ;
+      std::vector<double> coefRA;
+      std::vector<double> coefDEC;
+      std::vector<double> coefTWI;
+
+      int nPositionCoefficients = m_solveSettings->numberCameraPositionCoefficientsSolved();
+      int nPointingCoefficients = m_solveSettings->numberCameraAngleCoefficientsSolved();
+
+      // Indicate if we need to obtain default position or pointing values
+      bool useDefaultPosition = false;
+      bool useDefaultPointing = false;
+      // Indicate if we need to use default values when not solving twist
+      bool useDefaultTwist = !(m_solveSettings->solveTwist());
+
+      // If we aren't solving for position, set the number of coefficients to 1 so we can output the
+      // instrumentPosition's center coordinate values for X, Y, and Z
+      if (nPositionCoefficients == 0) {
+        nPositionCoefficients = 1;
+        useDefaultPosition = true;
+      }
+      // If we arent' solving for pointing, set the number of coefficients to 1 so we can output the
+      // instrumentPointing's center angles for RA, DEC, and TWI
+      if (nPointingCoefficients == 0) {
+        nPointingCoefficients = 1;
+        useDefaultPointing = true;
+      }
+
+      // Force number of position and pointing parameters to each be 3 (X,Y,Z; RA,DEC,TWI)
+      // so we can always output a value for them
+      int nPositionParameters = 3 * nPositionCoefficients;
+      int nPointingParameters = 3 * nPointingCoefficients;
+      int nParameters = nPositionParameters + nPointingParameters;
+
+      coefX.resize(nPositionCoefficients);
+      coefY.resize(nPositionCoefficients);
+      coefZ.resize(nPositionCoefficients);
+      coefRA.resize(nPointingCoefficients);
+      coefDEC.resize(nPointingCoefficients);
+      coefTWI.resize(nPointingCoefficients);
```

Comment:
Implementing these as QVectors removes the need to resize them and you can make simple appends when they are added to further down...

Reply:
Implemented as QVectors.

Reply:
@SgStapleton If I'm reading this correctly they are initialized empty. In which case they should be resized if the size is known ahead of time, otherwise you make a lot of unnecessary reallocs.

Reply:
@Kelvinrr I believe that is correct.